modules/runtime/nuxeo-runtime-migration/src/main/java/org/nuxeo/runtime/migration/MigrationServiceImpl.java

```diff
@@ -529,4 +533,55 @@ protected void atomic(String id, Consumer<KeyValueStore> consumer) {
         throw new RuntimeException("Cannot lock for write migration: " + id + ", already locked: " + currentLock);
     }
 
+    @Override
+    public Migration getMigration(String id) {
+        MigrationDescriptor descriptor = getDescriptor(XP_CONFIG, id);
+        return new Migration(id, descriptor.getDescription(), descriptor.getDescriptionLabel(), getStatus(id),
+                descriptor.getSteps());
+    }
+
+    @Override
+    public List<Migration> getMigrations() {
+        var descriptors = getDescriptors(XP_CONFIG);
+        return descriptors.stream().map(d -> getMigration(d.getId())).collect(Collectors.toList());
+    }
+
+    @Override
+    public void run(String migrationId) {
+        if(getStatus(migrationId) == null) {
+            probeAndSetState(migrationId);
```

Comment:
This method is synchronous and could take a lot of time as explained in its Javadoc. This could be problematic when using it through REST.

Reply:
Yes that may be why when I run and directly probe, I get an old result if I don't manually wait

Reply:
We agreed to have the `#run` method probing the migration to have only one call to do from the outside. However, like in the `#probeAndSetState` method Javadoc, it should be stated clearly in the `#run` method Javadoc.

Reply:
now probeAndRun with javadoc:
```
     /**
     * Probes the migration state if necessary and runs a migration.
     * THIS METHOD MAY TAKE A WHILE as it may call {@link #probeAndSetState(String)}.
     *
     * @since 11.1
     */
```